### DISCUSSION TITLE
Possible build for Windows 11 arm64?

### DISCUSSION THREAD

**Discussion by hathlife** [nillerusr/source-engine#285] [General]
has anybody successfully built playable artifact for windows 11 arm64 platform?

**Comment by kristibektashi** [comment]
Don't think so, but at some point someone was trying to port it to Windows ARM32 [(a real thing btw)](https://open-rt.gitbook.io/open-surfacert/common/windows-10), and they might have also done some work on ARM64: https://github.com/nillerusr/source-engine/pull/260.  If not, it could be a good place to start for anyone looking to implement this.

The ARM32 version itself would also run on any ARM64 Windows 10 or Windows 11 version BEFORE version 24H2, assuming the hardware it is being ran on supports ARMv7 applications. Windows 11 version 24H2 entirely removed ARM32 support from the kernel so that no ARM32 applications can run even if the CPU supports it.

**Reply by er2off** [reply]
Hey, author of winarm32 port is here. When I made this port, I could run it on my Lumia 930 but can't control player (lol, even with keyboard), so sersoft tested on its ARM64 Microsoft Surface and it worked too (of course it was pre-24H2)
But physics worked kinda incorrectly (I won't be surprised if it's MSVC moment because I already had one (7f267853f4529fba0a19855dc2a741496c89867a))

Speaking about native ARM64 port. It would be actually nice but I have no devices to test and I'm currently not interested in Windows support + I'm kinda lazy. And yea, actually some of ARM64 stuff was added into that PR even while it won't build as-is.

**Reply by kristibektashi** [reply]
Windows 10 ARM32 is kinda weird in general because it's an internal beta, yet I wonder how well it would work on Windows RT since that's a full release of the operating system, even if more limited, and can be jailbroken to run regular apps as well. Of course that would require general Windows 8.1 support as well, and I haven't checked if this version of the engine supports anything before Windows 10.

**Reply by kristibektashi** [reply]
Also https://github.com/mstorsjo/llvm-mingw is a Clang-based compiler that supports both ARM32 and ARM64 Windows, so perhaps you could test with it instead of MSVC, in order to see if it's MSVC's fault or just Windows 10 ARM32 being weird.

**Reply by er2off** [reply]
> Also https://github.com/mstorsjo/llvm-mingw is a Clang-based compiler that supports both ARM32 and ARM64 Windows

Yea, I already ported some apps from leaked winxp code using llvm-mingw, but... Source engine built with llvm-mingw crashes even on x64 (and yes, it also needs code modifications to just build! IDK why I didn't upload these changes on GitHub...), so it needs more debugging (currently I gave up and just build with msvc)

About Windows RT support: I think it will work as Source don't use any API above... WinXP? Win7? Anyway d3d9.dll should exist and other libs should exist as well, unlike d3dx9.dll (M$ removed it from ARM builds and UWP because deprecated, I took it from Wine). BTW d3dx9 can be completely removed and replaced with functions from togl (I noticed that when I wanted to add DXVK)

**Comment by mittorn** [comment]
Also, there was leaked win8 beta arm builds which still use win7 interface and does not have winRT limitations. You may try running it on qemu-woa first to debug win32arm builds
